overlord/devicestate/systems.go

```diff
@@ -0,0 +1,169 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package devicestate
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/seed"
+)
+
+func checkSystemRequestConflict(st *state.State, systemLabel string) error {
+	st.Lock()
+	defer st.Unlock()
+
+	var seeded bool
+	if err := st.Get("seeded", &seeded); err != nil && err != state.ErrNoState {
+		return err
+	}
+	if seeded {
+		// the system is fully seeded already
+		return nil
+	}
+
+	// inspect the current system which is stored in modeenv, were are
```

Comment:
note, typo this should be

```
	// inspect the current system which is stored in modeenv, note we are
```

Reply:
Thanks for spotting this, I've opened https://github.com/snapcore/snapd/pull/8712